lib/api/core/models/repositories/repository.js

```diff
@@ -388,6 +395,71 @@ class Repository {
     return Object.assign({}, o);
   }
 
+  /**
+   * Recursively delete all objects in repository with a scroll
+   *
+   * @param {object} options - ES options (refresh)
+   * @param {object} part
+   * @returns {Promise<integer>} total deleted objects
+   */
+  truncate (options, part = null) {
+    if (part === null) {
+      return this.search({}, { scroll: '10m', size: 100 })
+        .then(objects => {
+          return this._truncatePart(objects, options)
+            .then(deletedObjects => {
+
+              if (objects.hits.length < objects.total) {
+                return this.truncate(options, {
+                  total: objects.total,
+                  deleted: objects.hits.length,
+                  scrollId: objects.scrollId
+                }).then(total => deletedObjects + total);
```

Comment:
I have to manually calculate the number of deleted objects because for `Profile` and `Role` repositories I have to take in account the protected objects that can't be deleted (`admin`, `default` and `anonymous`)